src/optim/bundle_adjustment.cc

```diff
@@ -261,9 +261,70 @@ bool BundleAdjuster::Solve(Reconstruction* reconstruction) {
 
   problem_.reset(new ceres::Problem());
 
+  if (options_.use_prior_motion) {
+    std::vector<double> vini_err;
+    std::vector<Eigen::Vector3d> src, dst;
+    vini_err.reserve(config_.NumImages());
+    src.reserve(config_.NumImages());
+    dst.reserve(config_.NumImages());
+    for (const auto image_id : config_.Images()) {
+      auto& image = reconstruction->Image(image_id);
+      if (image.HasTvecPrior()) {
+        src.push_back(image.ProjectionCenter());
+        dst.push_back(image.TvecPrior());
+        vini_err.push_back(
+            (image.ProjectionCenter() - image.TvecPrior()).norm());
+      }
+    }
+
+    std::cout << "\n Initial Alignment tvec err (median / mean / std): "
+              << Median(vini_err) << " / " << Mean(vini_err) << " / "
+              << StdDev(vini_err) << "\n";
+
+    if (src.size() > 3) {
+      SimilarityTransform3 tform;
+      // bool success = reconstruction->Align(src, dst, &tform);
+      RANSACOptions ransac;
+      ransac.max_error = (options_.motion_prior_xyz_std * 3.).norm();
+      const int min_num_inliers = 3;
+      bool success = reconstruction->AlignRobust(src, dst, min_num_inliers,
```

Comment:
I believe that this call is problematic if all vectors in `dst` are equal, because then an attempt may be made within the function to invert a non-invertible matrix.